### PR TITLE
strategy: add audacity maker

### PR DESCRIPTION
**Pull request by austin362667**
## Raw Idea
- [ ] Order Aggressiveness
  - Order Aggressiveness refers to the percentage of orders that are submitted at market prices, as opposed to limit prices.
  - How to observe from public market data stream?
- [X] Order Flow
  - Order flow is the difference between buyer-initiated and seller-initiated trading volume or number of trades.
    - use queue to do time-series rolling
    - accumulating trading volume for `orderFlowSize`
    -  counting trades of number for `orderFlowNumber`
---
## Data Processing
1.  Scale the series onto [-1, 1] with a Min-Max scaler. => trivial fit here!
2. Convert the scaled time series into polar coordinates. => preserves temporal dependency!

### CODE REVIEW

**Review by c9s — APPROVED** [review]

File: pkg/strategy/ktrade/minute.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+func (s *Minute) Bind(session *bbgo.ExchangeSession, orderExecutor *bbgo.GeneralOrderExecutor) {
+	s.session = session
+	s.orderExecutor = orderExecutor
+
+	position := orderExecutor.Position()
+	symbol := position.Symbol
+	s.StreamBook = types.NewStreamBook(symbol)
+	s.StreamBook.BindStream(session.MarketDataStream)
+
+	//store, _ := session.MarketDataStore(symbol)
```
Comment (c9s) [inline pkg/strategy/ktrade/minute.go:50]: clean up debug comment?

File: pkg/strategy/ktrade/minute.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+			newAskPrice := s.midPrice.Mul(fixedpoint.NewFromFloat(0.25)).Add(trade.Price.Mul(fixedpoint.NewFromFloat(0.75)))
+			log.Infof("short @ %f", newAskPrice.Float64())
+			if trade.Price.Compare(newAskPrice) > 0 {
+				s.placeOrder(context.Background(), types.SideTypeSell, s.Quantity, trade.Price.Mul(fixedpoint.NewFromFloat(1.001)), symbol)
+			} else {
+				s.placeOrder(context.Background(), types.SideTypeSell, s.Quantity, newAskPrice.Round(2, 1), symbol)
+			}
+
+		} else if trade.Side == types.SideTypeSell && trade.Price.Compare(s.midPrice) < 0 {
+			_ = s.orderExecutor.GracefulCancel(context.Background())
```
Comment (c9s) [inline pkg/strategy/ktrade/minute.go:70]: add error check?

File: pkg/strategy/ktrade/minute.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+		log.Info(kline.NumberOfTrades)
+
+	}))
+
+	go func() {
+		quoteTicker := time.NewTicker(util.MillisecondsJitter(time.Millisecond*10, 200))
+		defer quoteTicker.Stop()
+
+		for {
+			select {
```
Comment (c9s) [inline pkg/strategy/ktrade/minute.go:94]: add `case <-ctx.Done():` case?

File: config/audacitymaker.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  binance:
+    exchange: binance
+    envVarPrefix: binance
+#    futures: true
+
+exchangeStrategies:
+- on: binance
+  audacitymaker:
+    symbol: ETHBUSD
+    pertrade:
```
Comment (c9s) [inline config/audacitymaker.yaml:19]: perTrade?

**Review by c9s — APPROVED** [review]

File: config/audacitymaker.yaml (hunk 4 above)
Comment (austin362667) [reply config/audacitymaker.yaml]: sure

Comment (austin362667) [reply config/audacitymaker.yaml]: done

**Review by andycheng123 — APPROVED** [review]

**Pull request merged**